examples/src/main/kotlin/com/teamdev/jxbrowser/examples/JarProtocol.kt

```diff
@@ -0,0 +1,126 @@
+import com.teamdev.jxbrowser.net.UrlRequestJob
+import com.teamdev.jxbrowser.net.callback.InterceptUrlRequestCallback
+import com.teamdev.jxbrowser.net.callback.InterceptUrlRequestCallback.*
+import com.teamdev.jxbrowser.net.callback.InterceptUrlRequestCallback.Response.intercept
+import com.teamdev.jxbrowser.net.callback.InterceptUrlRequestCallback.Response.proceed
+import java.net.URL
+import java.net.URLConnection
+
+/**
+ * This example demonstrates how to register a JAR protocol handler
+ * to load files located inside a JAR archive.
+ */
+fun main() {
+    // Create an engine with the custom JAR requests interceptor.
+    val engine = Engine(RenderingMode.HARDWARE_ACCELERATED) {
+        schemes.add(Scheme.JAR, InterceptJarRequestCallback())
+    }
+
+    val browser = engine.newBrowser()
+
+    // Load `index.html` file located inside `tiny-website.jar` added
+    // to this Java app runtime classpath.
+    val website = object {}.javaClass.getResource("/resources/index.html")!!
+    browser.navigation.loadUrl("$website")
+
+    singleWindowApplication(title = "JAR Protocol Handler") {
+        BrowserView(browser)
+    }
+}
+
+/**
+ * Overrides response data for requests that ask for a file from Jar archives.
+ *
+ * Such interceptor is needed when Chromium requests files from Jar archives.
+ * Such requests take place when an initially loaded page itself is located
+ * in the application resources (and thus, in a Jar archive).
+ *
+ * This page may initiate loading of other resources: style sheets, images,
+ * scripts, etc. And they all are located in a Jar just like `index.html`.
+ * Their paths are usually relative. Chromium can't load them on its own.
+ *
+ * This class intercepts such requests, loads the files on its own
+ * and overrides response data.
+ */
+private class InterceptJarRequestCallback : InterceptUrlRequestCallback {
+
+    override fun on(params: Params): Response {
+
+        // Check if the requested URL points to a file inside a Jar archive.
+        val request = params.urlRequest()
+        if (!request.url().startsWith("jar:")) {
+            // Do nothing if not.
+            return proceed()
+        }
+
+        // Read the requested file from the resources.
+        val javaUrl = request.toJavaUrl()
+        val entry = JarEntry(javaUrl)
+
+        // Override response data.
+        val job = overrideResponse(entry, params)
+        return intercept(job)
+    }
+
+    private fun overrideResponse(
+        entry: JarEntry,
+        params: Params
+    ): UrlRequestJob {
+        val options = UrlRequestJobOptions(HttpStatus.OK, entry.mimeType)
+        val job = params.newUrlRequestJob(options).apply {
+            write(entry.data)
+            complete()
+        }
+        return job
+    }
+}
+
+/**
+ * Converts this [UrlRequest] from Chromium notation
+ * to [Java's one][java.net.JarURLConnection].
+ *
+ * In particular, `jar://file/path` is converted to `jar:file:/path`.
```

Comment:
Here, it's the only conversion the method does.

I've clarified the first sentence here, so that it's `Converts  this [UrlRequest] for a JAR file ...`.